src/ray/object_manager/test/pull_manager_test.cc

```diff
@@ -0,0 +1,160 @@
+
+#include "ray/object_manager/pull_manager.h"
+
+#include "gtest/gtest.h"
+#include "ray/common/test_util.h"
+
+namespace ray {
+
+class PullManagerTest : public ::testing::Test {
+ public:
+  PullManagerTest()
+      : self_node_id_(NodeID::FromRandom()),
+        object_is_local_(false),
+        num_send_pull_request_calls_(0),
+        rand_int_(0),
+        num_restore_spilled_object_calls_(0),
+        pull_manager_(self_node_id_,
+                      [this](const ObjectID &object_id) { return object_is_local_; },
+                      [this](const ObjectID &object_id, const NodeID &node_id) {
+                        num_send_pull_request_calls_++;
```

Comment:
It would be nice to check something about which node IDs are requested. Something like, "all node IDs are requested once" (I'm not sure what the actual invariant is).

Reply:
So I would love to add an invariant like that, but I don't think the current behavior does that. If the timer ticks multiple times, it is possible to send multiple pull requests to the same worker (not sure if that's a feature or bug tbh...)

Reply:
Btw please let me know if there's some invariant that I just don't see right now.

Reply:
Is there an invariant when the timer ticks multiple times for a set of locations (and the set doesn't change in between)?

Reply:
I don't think so. We just do a simple random sample (with replacement). IMO this is one of the first low hanging fruit that we should deal with though.

Reply:
Gotcha, thanks for explaining!